CI: ci-runtime: Test_MultiPoolManager_ReleaseUnusedCIDR_PreAlloc timeout (v1.17)

## CI failure

```
panic: test timed out after 10m0s
running tests:
    Test_MultiPoolManager_ReleaseUnusedCIDR_PreAlloc (9m58s)

goroutine 498 [chan send]:
github.com/cilium/cilium/pkg/ipam.(*fakeK8sCiliumNodeAPIResource).updateNode(...)
    pkg/ipam/multipool_test.go:723

goroutine 500 [chan receive]:
github.com/cilium/cilium/pkg/ipam.(*multiPoolManager).ciliumNodeEventLoop(...)
    pkg/ipam/multipool.go:266
```

--- a/pkg/ipam/multipool.go
+++ b/pkg/ipam/multipool.go
@@ -477,7 +477,8 @@ func (m *multiPoolManager) isRestoreFinishedLocked(family Family) bool {
 
 func (m *multiPoolManager) updateCiliumNode(ctx context.Context) error {
 	m.mutex.Lock()
-	newNode := m.node.DeepCopy()
+	currentNode := m.node.DeepCopy()
+	newNode := currentNode.DeepCopy()
 	requested := []types.IPAMPoolRequest{}
 	allocated := []types.IPAMPoolAllocation{}
 
@@ -546,7 +547,7 @@ func (m *multiPoolManager) updateCiliumNode(ctx context.Context) error {
 
 	m.mutex.Unlock()
 
-	if !newNode.Spec.IPAM.DeepEqual(&m.node.Spec.IPAM) {
+	if !newNode.Spec.IPAM.DeepEqual(&currentNode.Spec.IPAM) {
 		_, err := m.nodeUpdater.Update(ctx, newNode, metav1.UpdateOptions{})
 		if err != nil {
 			return fmt.Errorf("failed to update node spec: %w", err)

